src/control/server/mgmt_pool.go

```diff
@@ -511,8 +551,30 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 		return nil, err
 	}
 	req.SetUUID(uuid)
+	req.SvcRanks = system.RanksToUint32(ps.Replicas)
 
 	resp := &mgmtpb.PoolDestroyResp{}
+
+	// If recursive flag is unset, refuse to destroy pool if resident containers exist.
+	if !req.Recursive {
+		hasContainers, lcRespStatus, err := svc.poolHasContainers(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		lcStatus := daos.Status(lcRespStatus)
+
+		if lcStatus != daos.Success {
+			svc.log.Errorf("ListContainers during pool destroy failed: %s", lcStatus)
+			resp.Status = lcRespStatus
+
+			return resp, nil
+		}
```

Comment:
`poolHasContainers` should probably translate the DAOS status int to an error if needed before returning. In the evict code it seems like there's a need for the status to make some other decisions, but here that's not the case.

Reply:
I've implanted as is to keep the 2 helper functions in some sort of consistency, changing poolHasContainers() to return an error if resp.Status != 0 would seem odd because then you would have to also communicate the Status value alongside the error in order to populate resp.Status. Do you feel strongly enough about this to require a change and repush? I don't really want to add to CI load unnecessarily.

Reply:
I could pass the resp by reference into poolHasContainers() but I feel that makes the intent of the function less clear.

Reply:
Ah, that makes sense. I wasn't really thinking about the status going into the resp as well.